doc/api/deprecations.md

```diff
@@ -3019,6 +3019,22 @@ should have the same effect. The receiving end should also check the
 [`readable.readableEnded`][] value on [`http.IncomingMessage`][] to get whether
 it was an aborted or graceful destroy.
 
+### DEP0157: Thenable support in streams
+
+<!-- YAML
+changes:
+  - version: REPLACEME
+    pr-url: https://github.com/nodejs/node/pull/40860
+    description: Documentation-only deprecation.
+-->
+
+Type: Documentation-only
+
+An undocumented feature of Node.js streams was to support thenables in
+implementation methods. This feature was causing stream instability when using
+async functions as stream methods. This is now deprecated, instead use
+callbacks.
+
```

Comment:
```suggestion

This feature caused users to encounter unexpected problems where the user
implements the function in callback style but uses e.g. an async method which
would cause an error.
```js
const w = new Writable({
  async final (callback) {
    await someOp(),
    callback(); // Not valid
  }
});
```